doc/content/getting-started/migrating-from-networks/device-json.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Create Device.json"
+description: ""
+weight: 10
+---
+
+{{% tts %}} allows you to import devices from other networks using a JSON file describing those devices. Devices imported this way can be migrated without the need for a rejoin.
+
+## Required Fields in `devices.json`
+
+| Field | Type | Description |
+|---|---|---|---|
+| `device_id` | string | [More info]({{< ref "reference/glossary#device-id" >}}) |
+| `application_id` | string | [More info]({{< ref "reference/glossary#application-id" >}}) |
+| `dev_eui` | uint64 | [More info]({{< ref "reference/glossary#deveui" >}}) |
+| `join_eui` | uint64 | Also referred to as **AppEUI**. [More info]({{< ref "reference/glossary#joineui" >}}) |
+| `name` | string | Optional, name of the device |
+| `description` | string | Optional, description of the device |
+| `lorawan_version` | `defined_only` | e.g.  MAC_V1_0_2`. [More info]({{< ref "reference/glossary#lorawan-version" >}}) |
+| `lorawan_phy_version` | `defined_only` | e.g.  PHY_V1_0_2_REV_B`. Also referred to as **Regional Parameters Version**. [More info]({{< ref "reference/glossary#regional-parameters" >}}) |
+| `frequency_plan_id` | `defined_only` | e.g.  EU_863_870`. [More info]({{< ref "reference/glossary#frequency-plan" >}}) |
+| `supports_join` | boolean | `true` for OTAA, `false` for ABP devices |
+| `app_key` | uint128 | Application Key. [More info]({{< ref "reference/glossary#application-key" >}}) |
+| `nwk_key` | uint128 | Network Key. Only for LoRaWAN version 1.1+ |
```

Comment:
`nwk_key` is not a top-level field in the JSON file, so this might become confusing for users. The example JSON below *should* help sort most of that out.

However, I would suggest that the field names here are changed to imply the "full" path. e.g. change `nwk_key` to `root_keys.nwk_key.key`, `device_id` to `ids.device_id` etc. Maybe also add a comment for the meaning of the dot notation.

Adding the comment under `nwk_key`, but this obviously applies to most of the keys here.

Reply:
@neoaggelos not sure if I understand your comment.

What do you mean with *top-level field*?

Why would someone rather use `root_keys.nwk_key.key` instead of `nwk_key`, or `ids.device_id` instead of `device_id`? 

And, what is the meaning of the dot notation?

Reply:
The problem is that a user that sees `nwk_key` in the table above, they may be misled to believe that the JSON file should have the following format:

```js
{
  "ids": {
    "device_id": "dev1",
     /* .... */
   },
  "nwk_key": "<NWK_KEY_HERE>",
  /* .... */
}
```

However, that is wrong, and they should instead do:

```json
{
  "ids": {
    "device_id": "dev1",
    /* .... */
  },
  "root_keys": {
    "nwk_key:": {
      "key": "<NWK_KEY_HERE>"
    }
  },
  /* ... */
}
```

The "dot notation" I refer to specifies the full path to the key. Hope this makes things a bit more clear.

Reply:
Thanks, this clarifies.